src/components/BillingRemaining.tsx

```diff
@@ -0,0 +1,157 @@
+import { IS_NIGHTLY_OR_DEBUG } from '@src/routes/utils'
+import { Spinner } from '@src/components/Spinner'
+import { useSelector } from '@xstate/react'
+import { useState } from 'react'
+
+import { CustomIcon } from '@src/components/CustomIcon'
+import type { BillingActor } from '@src/machines/billingMachine'
+
+export enum BillingRemainingMode {
+  ProgressBarFixed,
+  ProgressBarStretch,
+}
+
+export interface BillingRemainingProps {
+  mode: BillingRemainingMode
+  billingActor: BillingActor
+}
+
+const Error = (props: { error: Error }) => {
+  const [showMessage, setShowMessage] = useState(false)
+
+  const fadedBg = 'rgba(127, 127, 127, 0.2)'
+  const fadedFg = 'rgba(255, 255, 255, 0.6)'
+  const colors = {
+    color: fadedFg,
+    stroke: fadedFg,
+    fill: fadedFg,
+    backgroundColor: fadedBg,
+  }
+
+  return (
+    <div
+      onMouseEnter={() => setShowMessage(true)}
+      onMouseLeave={() => setShowMessage(false)}
+    >
+      {showMessage && (
+        <div
+          style={{
+            position: 'absolute',
+          }}
+        >
+          <div
+            className="rounded p-1"
+            style={{ ...colors, position: 'relative', top: -32 }}
+          >
+            {props.error.toString()}
+          </div>
+        </div>
+      )}
+      <div className="rounded" style={colors}>
+        <CustomIcon name="exclamationMark" className="w-5 h-5" />
+      </div>
+    </div>
+  )
+}
```

Comment:
Could you show a screenshot of what this looks like? Thanks for making an error state even though it was missing design! It kinda looks like it could be replaced with a `Tooltip`? But that can be later